### ISSUE TITLE
Error: Device check is not supported by this device

### ISSUE THREAD

**Issue by RamanPndy** [react-native-community/react-native-device-info#927]
# Bug

## Environment info

React native info output:

```bash
Ramans-MacBook-Pro:SocialBird ramanpandey$ react-native info
info Fetching system and libraries information...
System:
    OS: macOS Mojave 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
    Memory: 46.23 MB / 8.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 10.16.0 - /usr/local/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
  SDKs:
    iOS SDK:
      Platforms: iOS 12.4, macOS 10.14, tvOS 12.4, watchOS 5.3
  IDEs:
    Android Studio: 3.4 AI-183.6156.11.34.5692245
    Xcode: 10.3/10G8 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.5 => 0.61.5 
  npmGlobalPackages:
    create-react-native-app: 2.0.2
    react-native-cli: 2.0.1
```

Library version: 5.4.0

## Steps To Reproduce

```javascript
import DeviceInfo from 'react-native-device-info';
getDeviceInfo = async() =>{
    let os = await DeviceInfo.getBaseOs();
    let brand = DeviceInfo.getBrand();
    let buildId = await DeviceInfo.getBuildId();
    let buildNumber = DeviceInfo.getBuildNumber();
    let batteryLevel = await DeviceInfo.getBatteryLevel();
    let deviceId = DeviceInfo.getDeviceId();
    let deviceType = DeviceInfo.getDeviceType();
    let deviceName = DeviceInfo.getDeviceName();
    let deviceToken = await DeviceInfo.getDeviceToken();
    let firstInstallTime = await DeviceInfo.getFirstInstallTime();
    let lastUpdateTime = await DeviceInfo.getLastUpdateTime();
    let carrier = await DeviceInfo.getCarrier();
    let model = DeviceInfo.getModel();
    let ipAddress = await DeviceInfo.getIpAddress();
    let macAddress = await DeviceInfo.getMacAddress();
    let manufacturer = await DeviceInfo.getManufacturer();
    let systemName = DeviceInfo.getSystemName();
    let systemVersion = DeviceInfo.getSystemVersion();
    let uniqueId = DeviceInfo.getUniqueId();
    let userAgent = await DeviceInfo.getUserAgent();
    let readableVersion = DeviceInfo.getReadableVersion();
    let totalDiskCapacity = await DeviceInfo.getTotalDiskCapacity();
    let freeDiskStorage = await DeviceInfo.getFreeDiskStorage();
    let isEmulator = await DeviceInfo.isEmulator();
    let isLandscape = await DeviceInfo.isLandscape();
    let isTablet = DeviceInfo.isTablet();

    let payload = {
      "os": os,
      "brand" :  brand,
      "buidId" : buildId,
      "buildNumber" :  buildNumber,
      "batteryLevel" :  batteryLevel,
      "deviceId" :  deviceId,
      "deviceType": deviceType,
      "deviceName": deviceName,
      "deviceToken" : deviceToken,
      "firstInstallTime" :  firstInstallTime,
      "lastUpdateTime" : lastUpdateTime,
      "carrier" : carrier,
      "model" : model,
      "ipAddress" : ipAddress,
      "macAddress" : macAddress,
      "manufacturer" : manufacturer,
      "systemName" : systemName,
      "systemVersion" : systemVersion,
      "uniqueId" : uniqueId,
      "userAgent" :  userAgent,
      "readableVersion" :  readableVersion,
      "totalDiskCapacity" :  totalDiskCapacity,
      "freeDiskStorage" : freeDiskStorage,
      "isEmulator" : isEmulator,
      "isLandscape" :  isLandscape,
      "isTablet" : isTablet
    }

    console.log(payload)
  }
```

on call of this.getDeviceInfo()

I got following output:
```
Error: Device check is not supported by this device
fn@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2163:45
getDeviceToken$@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:115032:87
tryCatch@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29288:23
invoke@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29463:32
tryCatch@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29288:23
invoke@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29364:30
http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29394:19
tryCallTwo@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:3251:9
doResolve@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:3415:25
Promise@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:3274:14
callInvokeWithMethodAndArg@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29393:29
enqueue@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29398:157
async@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29414:69
getDeviceToken@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:115023:38
_callee2$@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:95709:96
tryCatch@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29288:23
invoke@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29463:32
tryCatch@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29288:23
invoke@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29364:30
http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29374:21
tryCallOne@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:3242:16
http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:3343:27
_callTimer@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:30617:17
_callImmediatesPass@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:30653:19
callImmediates@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:30872:33
callImmediates@[native code]
__callImmediates@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2652:35
http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2429:34
__guard@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2635:15
flushedQueue@http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2428:21
flushedQueue@[native code]
invokeCallbackAndReturnFlushedQueue@[native code]
```
1.
2.
...

Describe what you expected to happen:

1.I should get the Device Information
2.

**Comment by mikehardy** [comment]
Hi there!
getDeviceToken is only available for iOS11+ - https://github.com/react-native-community/react-native-device-info/blob/master/README.md#getdevicetoken

And the code guards it https://github.com/react-native-community/react-native-device-info/blob/master/ios/RNDeviceInfo/RNDeviceInfo.m#L380

@AntoineDoubovetzky can you think of any reason this could happen?

**Comment by mikehardy** [comment]
For reference I just attempted to reproduce this on my iOS9 tablet, and it worked fine (or I should say it did not crash, there was no token available of course)

**Comment by mikehardy** [comment]
Okay, I reproduce this on the simulator. If you use a Simulator at iOS13+ it will throw like this:

` LOG  Trouble getting device info  [Error: Device check is not supported by this device]`

So I suppose it needs a simulator check, similar to the battery level items

**Comment by mikehardy** [comment]
released in 5.4.1 :rocket:
